modules/porous_flow/src/materials/PorousFlowHysteresisOrder.C

```diff
@@ -0,0 +1,195 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowHysteresisOrder.h"
+#include "Conversion.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowHysteresisOrder);
+
+InputParameters
+PorousFlowHysteresisOrder::validParams()
+{
+  InputParameters params = PorousFlowMaterial::validParams();
+  params.addPrivateParam<std::string>("pf_material_type", "hysteresis_order");
+  params.addParam<unsigned>("liquid_phase", 0, "The phase number of the liquid phase");
+  params.addParam<unsigned>(
+      "initial_order",
+      0,
+      "The initial order.  0 means the simulation will begin on the primary drying curve.  1 means "
+      "the simulation will begin on the first-order wetting curve.  2 means the simulation will "
+      "begin on the second-order drying curve, etc.");
```

Comment:
Is there a limit to the order? If so, should you use a MooseEnum to restrict the order to certain values?

Reply:
Yes, the initial order must be <= PorousFlowConstants::MAX_HYSTERESIS_ORDER,  and i check this further down in the code.  I tried a RangeCheckedParam but that didn't work.